freud/environment.pyx

```diff
@@ -582,9 +582,9 @@ cdef class MatchEnv(Compute):
         points = freud.common.convert_array(points, shape=(None, 3))
         ref_points = freud.common.convert_array(ref_points, shape=(None, 3))
 
-        cdef np.ndarray[float, ndim=1] l_points = np.ascontiguousarray(
+        cdef const float[::1] l_points = np.ascontiguousarray(
             points.flatten())
-        cdef np.ndarray[float, ndim=1] l_ref_points = np.ascontiguousarray(
+        cdef const float[::1] l_ref_points = np.ascontiguousarray(
             ref_points.flatten())
         cdef unsigned int nP = l_points.shape[0]
         cdef unsigned int nRef = l_ref_points.shape[0]
```

Comment:
I'm going to revert these changes, the code here is actually completely wrong and won't work. The arrays have to be 2D for the C++ side. I fixed this in #489, so I'm just going to revert this part of it (nothing currently tests these functions, which is why we don't see it).

Reply:
Ah I see. The macros you added also make this code impossible. OK, I'm going to hold off on this PR until #489 is merged, then rebase this so that everything is happy.